posts/build-a-reddit-crawler-with-aws-cdk-and-lambda.md

```diff
@@ -0,0 +1,181 @@
+
+```typescript
+import * as cdk from '@aws-cdk/core';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as sns from '@aws-cdk/aws-sns';
+import * as subscriptions from '@aws-cdk/aws-sns-subscriptions';
+import * as events from '@aws-cdk/aws-events';
+import * as targets from '@aws-cdk/aws-events-targets';
+
+export class BlogRedditCrawlerStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    // Create SNS topic
+    const topic = new sns.Topic(this, 'RedditCrawler');
+
+    // Create lambda
+    const fn = new lambda.Function(this, 'fn', {
+      code: new lambda.AssetCode('resources'),
+      handler: 'lambda.handler',
+      runtime: lambda.Runtime.NODEJS_14_X,
+      functionName: 'reddit-crawler',
+      environment: {
+        TOPIC_ARN: topic.topicArn
+      },
+      timeout: cdk.Duration.seconds(30)
+    });
+
+    // Allow the lambda to publish to the topic
+    topic.grantPublish(fn);
+
+    // Add an email subscription to the topic
+    topic.addSubscription(new subscriptions.EmailSubscription('MY_EMAIL'));
+
+    // Trigger lambda every X minutes
+    new events.Rule(this, 'Rule', {
+      schedule: events.Schedule.rate(cdk.Duration.minutes(5)),
+      targets: [new targets.LambdaFunction(fn)]
+    });
+  }
+}
+```
+
+The stack code is pretty straightforward, don't forget to replace `MY_EMAIL` with your real email (you will have to verify your email and also confirm subscribing to the SNS topic).
+
+The SNS topic is the first component of the stack to be defined since the Lambda requires the variable `TOPIC_ARN` to be able to publish. The rest of the stack definition is pretty simple as we just have to use the correct functions to grant permissions between each components.
+
+The project is available at this [GitHub repository](https://github.com/Falydoor/blog-reddit-crawler) in case you want to clone it. You can follow the instructions to compile/deploy/destroy the stack:
+
+```bash
+# compile typescript to javascript
+npm run build
+
+# deploy the stack
+cdk deploy
+
+# destroy the stack
+cdk destroy
+```
+
+Your console should output something similar once the stack is successfully deployed:
+
+![Deploy](https://raw.githubusercontent.com/Falydoor/blog-usa/reddit-crawler/images/2021/02/reddit-crawler-deploy.png)
+
+After few minutes, you should receive an email looking like below:
+
+![Email](https://raw.githubusercontent.com/Falydoor/blog-usa/reddit-crawler/images/2021/02/reddit-crawler-email.png)
+
+Feel free to change the SNS message content in the Lambda to show more details about each post.
+
+# Pricing
+
+Only the Lambda and SNS are billable and our volume is very low because the Lambda runs every 30 minutes which is less than 1500 requests per month. Also, the Lambda runs with only 128MB of memory and each request take less than 5 seconds on average. Here is an estimation of how much this crawler will cost:
```

Comment:
```suggestion
Only the Lambda and SNS are billable, and our volume is very low because the Lambda runs every 30 minutes which is less than 1500 requests per month. Also, the Lambda runs with only 128MB of memory and each request takes less than 5 seconds on average. Here is an estimation of how much this crawler will cost:
```